packages/edit-site/src/components/header-edit-mode/document-actions/index.js

```diff
@@ -144,6 +145,13 @@ function TemplateDocumentActions( { className, onBack } ) {
 		typeIcon = symbol;
 	}
 
+	const { refererId } = getQueryArgs( window.location.href );
+
+	if ( ! onBack && refererId ) {
+		onBack = () =>
+			( document.location = `post.php?post=${ refererId }&action=edit` );
```

Comment:
Nit: Do we want to add some guard here so that arbitrary `refererId` won't get injected here?

Reply:
yeh, I was thinking it has to accept an arbitrary post id anyway and the `post.php` route should be sanitizing input, eg. you can manually add an arbitrary id in `/wp-admin/post.php?post=` anyway. But I have tidied it up so `refererId` at least needs to be a number. Let me know if you can see anything else we should be doing here.